packages/web3-eth-ens/src/ENS.js

```diff
@@ -33,21 +34,40 @@ var ResolverMethodHandler = require('./lib/ResolverMethodHandler');
  */
 function ENS(eth) {
     this.eth = eth;
-}
+    var registryAddress = null;
+    this._detectedAddress = null;
+    this._lastSyncCheck = null;
+
+    Object.defineProperty(this, 'registry', {
+        get: function () {
+            return new Registry(this);
+        },
+        enumerable: true
+    });
+
+    Object.defineProperty(this, 'resolverMethodHandler', {
+        get: function () {
+            return new ResolverMethodHandler(this.registry);
+        },
+        enumerable: true
+    });
 
-Object.defineProperty(ENS.prototype, 'registry', {
-    get: function () {
-        return new Registry(this);
-    },
-    enumerable: true
-});
+    Object.defineProperty(this, 'registryAddress', {
+        get: function () {
+            return registryAddress;
+        },
+        set: function (value) {
+            if (value === null) {
+                registryAddress = value;
```

Comment:
How does this setter work? Just curious ... I see the registryAddress being set in the tests and everything looks great.

Reply:
I've used the ``inputAddressFromatter`` to format and validate the ``address`` as we are doing on other points as well. With the exception of the additional valid value ``null``. The ``null`` value can be used to "clear" the custom-defined registry address and to "re-activate" the automatic registry search the ``ENS`` module does have.

Reply:
Ah thanks for the explanation.